File: DSA/Ayush_Mishra_590014844/Day_19_Question_2.py
def find_peak(arr):
    low, high = 0, len(arr) - 1
    while low < high:
        mid = (low + high) // 2
        if arr[mid] < arr[mid + 1]:
            low = mid + 1
        else:
            high = mid
    return low

arr = list(map(int, input("Enter array: ").split()))
print("Peak index:", find_peak(arr))

File: DSA/Ayush_Mishra_590014844/Day_19_Question_2.c
#include <stdio.h>

int find_peak(int arr[], int n) {
    int low = 0, high = n - 1;
    while (low < high) {
        int mid = (low + high) / 2;
        if (arr[mid] < arr[mid + 1])
            low = mid + 1;
        else
            high = mid;
    }
    return low;
}

int main() {
    int n;
    printf("Enter size of array: ");
    scanf("%d", &n);
    int arr[n];
    printf("Enter elements: ");
    for (int i = 0; i < n; i++) scanf("%d", &arr[i]);
    int peak = find_peak(arr, n);
    printf("Peak index: %d\n", peak);
    return 0;
}
